### ISSUE TITLE
Some ZIP files doesn't open with external viewer due Path Issue

### ISSUE THREAD

**Issue by NiceGlassesDude** [happypandax/happypandax#118]
I personally didn't quite understood why this is happening... here is an example.

I do add ZIP files for reading and some of them are registered with the correct path and they can be opened with the external viewer like this: {ext-viewer}/comic.zip. But some aren't and they get a weird path like this: {ext-viewer}/comic.zip/comic.zip. The path basically multiplies itself and I don't know why.

**Comment by twiddli** [comment]
Hello, could you run HPX server in debug mode, retry the action and then provide me your debug.log?
You can enable debug mode through the webclient in Perferences -> Advanced.

**Comment by NiceGlassesDude** [comment]
Hi twiddli, firstly here is what the console says:

```
Traceback (most recent call last):
  File "env\lib\site-packages\gevent\threadpool.py", line 281, in _worker
io.UnsupportedOperation: seek
2019-07-06T20:33:43Z (<ThreadPool at 0x10a5db0 0/1/10 hub=<Hub at 0xbd1500 thread_ident=0x13168>>, <built-in method seek of ZipExtFile object at 0x010A57F0>) failed with UnsupportedOperation
```
Here is my log: https://workupload.com/file/qEDaQsZ8

**Comment by twiddli** [comment]
I see that the path sent to the viewer has this part in it `\DnA 2.zip\DnA 2` or `\DnA 2.zip\DnA 2\DNA_2_00.png`, is this what you were talking about?

**Comment by NiceGlassesDude** [comment]
Yes. For some reason some zip's are working and others don't. But It works on HP itself so there you can view the pages. However speaking about that, there are also some files and they don't even show up on the MP front page. It's greyed out. Of course you can't open them in the external viewer either.

Aside from that this project you're working on is absolutely incredible. Thank you very much. If it works as intended I could even stack my normal comics in there (marvel etc.).

**Comment by twiddli** [comment]
Hi, it's happening because the image files are contained inside an additional folder inside the archive file.
HPX does no extraction prior to sending them off the the external viewer. This was something HP did with a setting. I will add this to HPX.

> However speaking about that, there are also some files and they don't even show up on the MP front page. It's greyed out. 

Could you expand on this? I don't understand what you mean.

**Comment by NiceGlassesDude** [comment]
I see there was the problem, thanks twiddli. With the second "issue" I meant the cover. If I scan as instance a folder with several manga, mostly everything works just fine which is awesome and MPX even shows their cover. However, there are also exceptions and it doesn't show their cover. I just checked their content and It's the same as the other ones.

**Comment by twiddli** [comment]
this should now be fixed in `v0.11.0` with the new extract option
